### PR TITLE
workflow protection for manual runs

### PR DESCRIPTION
**Pull request by mydesignrocks**
## Description
The environment within github for now is named as "dev-test" and this can be changed based on:
> successful execution of this
> and the cleanup of environment task doesnt delete anything that is other than master

Manual trigger of the workflow at deploy step will trigger an approval to the vsp-frontend-team members
This was also tested here to make sure the scheduled jobs are not required to go through an approval.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#37940

## Testing done
yes

## Acceptance criteria
- [X]

## Definition of done
- [X] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by timwright12** [comment]
Same question:
@mydesignrocks

> approval to the vsp-frontend-team members

Will members of vsp-frontend-team be able to deploy without approval? We'll need that for after hours stuff if no one else is around

**Comment by mydesignrocks** [comment]
> Same question: @mydesignrocks
> 
> > approval to the vsp-frontend-team members
> 
> Will members of vsp-frontend-team be able to deploy without approval? We'll need that for after hours stuff if no one else is around

missed out on this comment, sorry. The approval can be self approved... meaning, it triggers an approval but can be reviewed and approved by the same individual as long as the individual is part of this: 
image

File: .github/workflows/daily-deploy-production.yml
Comment (theodur) [inline .github/workflows/daily-deploy-production.yml:32]: Should we change the name of the environment to `production` now that we're not testing this anymore?

Also, we might not need the `else` statement that sets the output string as empty. If there's no output set for a variable, I think it defaults to an empty string. If that's the case, we can probably just use an `if` statement in GHA. Something like:
``` yaml
if: github.event_name == 'workflow_dispatch'
run: echo ::set-output name=env_name::'production'
```

Comment (mydesignrocks) [reply .github/workflows/daily-deploy-production.yml]: updated

**Comment by mydesignrocks** [comment]
@theodur - and to your comment on the `if` condition, my thought was - since it was declared as shell and not within the workflow action, the existing pattern is ok. Otherwise, I am thinking it might ignore the manual run completely. We can test if needed but wondering if we could start with this...

**Comment by theodur** [comment]
> @theodur - and to your comment on the `if` condition, my thought was - since it was declared as shell and not within the workflow action, the existing pattern is ok. Otherwise, I am thinking it might ignore the manual run completely. We can test if needed but wondering if we could start with this... 

Yeah, the current pattern we used when testing is fine. I was just noticing that this could be less verbose (I didn't refactor after we got this working). But we can start with this and test my suggestion later if you want

**Review by theodur — APPROVED** [review]

**Pull request merged**
